### DISCUSSION TITLE
Not able to set "Available to budget" amount

### DISCUSSION THREAD

**Discussion by MattDemers** [firefly-iii/firefly-iii#7660] [Firefly III] [answered]
Hey all; just discovering Firefly III today. Longtime YNAB4 user, and happy to move to something self-hosted.

For context, in YNAB4 I would add a starting amount into my accounts. It would then take this amount and make it available to be budgeted into buckets/envelopes, or "budget" items (Budgets in Firefly).

Looking at the documentation, I'm a bit confused by how to replicate this. From my understanding it doesn't do that automatically, and that's fine; [the documentation](https://docs.firefly-iii.org/firefly-iii/financial-concepts/organizing/) references that I should be able to do this manually:

> Each budget is valid for the current time period only (look to the top right). First, set the amount that is available to you by clicking the pencil icon next to "Available in \~\~\~". By default, this is zero.

However, on my /budgets screen, I don't seem to have the option, or see a pencil icon that I'm looking for.

![brave_WPk6aB73IR](https://github.com/firefly-iii/firefly-iii/assets/7098804/edd5b48c-7871-4918-bb9a-8c61cb09209a)

I created a bunch of budgets and assigned the amounts that I was using previously, but the first bar said "Your budgeted amounts is the same as your Available between June 1-30", which doesn't exactly help me. Am I missing something? Ideally I'd be able to set my "Available between June 1-30" to the amount in my checking account and then I'd be able to see if I budgeted more than I had available to me.

For reference, I have my Checking, Line of Credit and Credit Card created as Asset Accounts.

**Comment by JC5** [comment, marked as answer]
Ah, I will update that text. That amount can no longer be set. It's calculated from your budgets.

**Reply by MattDemers** [reply]
Cheers, thank you.

I know that Firefly doesn't work exactly like YNAB/YNAB4, mostly because the existence of Categories. YNAB wants me to "give every dollar a job" which means assigning every dollar to the equivalent of a Budget. Categories exist outside of that system, so it kind of breaks down.

I have Firefly set up so that my Net Worth is just the amount in my Checking, minus my Credit Card debt (if I have any). If I wanted to keep working with that YNABish system within Firefly, would I just make sure that the amount budgeted never exceeds my net worth?

I'm basically just looking for some way to limit the amount of money I'm budgeting, or that is available to budget, so I don't end up budgeting more than I can afford. Some kind of $ limit in the Budgeting UI, or some way to reference the amounts in my account, would really help.

Looking at the Piggy Banks UI, something similar would be great for Budgeting, which hopefully would cross-reference; "amount in Budgets", "remaining for Piggy Bank" etc.

![image](https://github.com/firefly-iii/firefly-iii/assets/7098804/77cb1a36-011a-4df8-b974-dfa854389904)

**Reply by JC5** [reply]
What I wrote about that is here: https://docs.firefly-iii.org/firefly-iii/more-information/personal-finances/

**Reply by MattDemers** [reply]
(Not trying to be a dick, but for anyone reading this afterward, I believe @JC5 meant to reference this link: https://docs.firefly-iii.org/firefly-iii/more-information/zero-based-budgeting/)

**Reply by JC5** [reply]
That one too lol 🫣

**Reply by gignsky** [reply]
I'm reading this a bit after it was posted but I'm still a little bit confused @JC5, how does firefly calculate the available between amount? From experimentation I can see that it seems to be summing the total of the budgeted column. But if this is the case what is the purpose of having the budgeted vs available to budget bar? Also how is this amount calculated if you are doing a budget on a per week basis but have some items that apply to the whole month?
